benchmarking/runs/mutations/mutation_benchmarking.py

```diff
@@ -0,0 +1,208 @@
+from benchmarking.simulation.simulation_set import SimulationSetArtifact, SimulationSet
+from benchmarking.simulation.simulation_settings import SimulationSettings
+
+
+class MutationBenchmarking(Run):
+    def start(self, num_trials: int = 100, generate_graphs: bool = False):
+        class_sizes = [40, 100, 240, 500, 1000]
+        team_size = 5
+
+        scenario = (
+            SatisfyProjectRequirementsAndDiversifyFemaleMinOf2AndDiversifyAfricanMinOf2()
+        )
+
+        metrics = {
+            "PrioritySatisfaction": PrioritySatisfaction(
+                goals_to_priorities(scenario.goals),
+                False,
+            ),
+            "AverageProjectRequirementsCoverage": AverageProjectRequirementsCoverage(),
+            "AverageCosineDifferenceGender": AverageCosineDifference(
+                attribute_filter=[ScenarioAttribute.GENDER.value],
+                name="AverageCosineDifferenceGender",
+            ),
+            "AverageCosineDifferenceRace": AverageCosineDifference(
+                attribute_filter=[ScenarioAttribute.RACE.value],
+                name="AverageCosineDifferenceRace",
+            ),
+            "AverageSoloStatus": AverageSoloStatus(
+                minority_groups_map={
+                    ScenarioAttribute.GENDER.value: [Gender.FEMALE.value],
+                    ScenarioAttribute.RACE.value: [Race.African.value],
+                },
+            ),
+            Insight.KEY_RUNTIMES: "runtime",
+        }
+
+        max_keep = 30
+        max_spread = 100
+        max_iterate = 250
+        max_time = 1_000_000
+        mutation_sets = {
+            "mutate_random": [RandomSwapMutation(max_spread)],
+            "mutate_local_max": [
+                LocalMaxMutation(num_mutations=1),
+                RandomSwapMutation(num_mutations=max_spread - 1),
+            ],
+            "mutate_local_max_random": [
+                LocalMaxRandomMutation(num_mutations=5),
+                RandomSwapMutation(num_mutations=(max_spread - 5)),
+            ],
+            "mutate_random_slice": [RandomSliceMutation(max_spread)],
+            "mutate_half_random_slice": [
+                RandomSliceMutation(num_mutations=(max_spread // 2)),
+                RandomSwapMutation(num_mutations=(max_spread // 2)),
+            ],
+            "mutate_greedy_local_max_n_2": [
+                GreedyLocalMaxMutation(number_of_teams=2, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_n_4": [
+                GreedyLocalMaxMutation(number_of_teams=4, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_n_8": [
+                GreedyLocalMaxMutation(number_of_teams=8, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_with_random_swap": [
+                GreedyLocalMaxMutation(num_mutations=(max_spread // 2)),
+                RandomSwapMutation(num_mutations=(max_spread // 2)),
```

Comment:
we're floor dividing a lot here, this gets funky if someone uses this code and messes with `max_spread` (i.e. making it an odd number), not really sure what the solution is though, maybe a comment? Maybe docs? This comment is just to illuminate the issue